src/sas/webfit/analyze/fitting/views.py

```diff
@@ -0,0 +1,272 @@
+from sas.qtgui.Utilities.CategoryInstaller import CategoryInstaller
+
+from serializers import (
+    FitSerializer,
+    FitParameterSerializer,
+)
+from .models import (
+    Fit,
+    FitParameter,
+)
+
+
+fit_logger = getLogger(__name__)
+model_manager = ModelManager()
+#TODO add view that gets previous fit results
+#TODO finish view for get status
+
+#start() only puts all the request data into the db, start_fit() runs calculations
+"""
+Documentation for user input: User input should look like this:
+{
+    "current_user": int
+    "data_id": int
+    "model":str
+    "parameters": [
+        {
+        "name":str
+        "value":int/float
+        "datatype":
+        "lower":int
+        "upper":int
+        }
+    ]
+}
+"""
+
+@api_view(['POST'])
+def start(request, version = None):
+    if request.method == "POST":
+        #TODO set status
+        #TODO add constraints
+        
+        base_serializer = FitSerializer(data=request.data)
+        if base_serializer.is_valid():
+            base_serializer.save()
+        else:
+            return Response(base_serializer.errors)
+        
+        fit_db = get_object_or_404(Fit, id = base_serializer.data["id"])
+
+        if not load_model(fit_db.model):
+            fit_db.delete()
+            return HttpResponseBadRequest("No model selected for fitting")
+
+        if fit_db.data_id:
+            if not (fit_db.data_id.is_public or (request.user.is_authenticated and 
+                    request.user is fit_db.data_id.current_user)):
+                fit_db.delete()
+                return HttpResponseBadRequest("data isn't public and/or the user's")
+
+        if request.data.get("parameters"):
+            parameters = request.data.get("parameters")
+            
+            #list of FitParameter objects
+            all_param_dbs = []
+            for x in parameters:
+                x["base_id"] = fit_db.id
+                parameter_serializer = FitParameterSerializer(data = x)
+                if parameter_serializer.is_valid():
+                    parameter_serializer.save()
+                all_param_dbs.append(get_object_or_404(FitParameter, base_id = fit_db.id, name = x["name"]))
+
+        #TODO move below data to created queuing function
+        result = start_fit(fit_db)
+
+        result_serializer = FitSerializer(fit_db, data = {"results": str(result), "status":3}, partial=True)
+        if result_serializer.is_valid():
+            result_serializer.save()
+        else:
+            return Response(result_serializer.errors)
+        fit_db = get_object_or_404(Fit, id = result_serializer.data["id"])
+        
+        #add "warnings": ... later
+        return Response({"authenticated":request.user.is_authenticated, "fit_id":fit_db.id, "results":result})
+
+    return HttpResponseBadRequest()
+
+
+def start_fit(fit_db):
+    #fit_db.status = 2
+    current_model = load_model(fit_db.model)
+
+    pars, par_limits = get_parameters(fit_db.id)
+
```

Comment:
Probably good to have default values (e.g. 0.0005) stored in settings file

Reply:
Ah yes, that's a good point. I can set the defaults in the models database, they're set to none right now